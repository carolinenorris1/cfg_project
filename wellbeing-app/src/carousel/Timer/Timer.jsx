import React, { useState } from "react";
import { CountdownCircleTimer } from "react-countdown-circle-timer";
import eyeAnimationSideToSide from "./timer-animations/eye-animation-side-to-side.gif";
import eyeAnimationLookingUp from "./timer-animations/eye-animation-looking-up.gif";
import birdSound from "./timer-sounds/birds-singing-trimmed.mp3";
import cheeringCrowdSound from "./timer-sounds/crowd-cheer-trimmed.mp3";
import alarmClockSound from "./timer-sounds/mechanical-alarm-clock-trimmed.mp3";
import pianoSound from "./timer-sounds/the-last-piano-trimmed.mp3";
import motivationalMusicSound from "./timer-sounds/motivational-trimmed.mp3";
import soothingMusicSound from "./timer-sounds/cottagecore-trimmed.mp3";

function EyeTimer() {
  const [key, setKey] = useState(0);
  const [runTimeInMinutes, setRunTimeInMinutes] = useState(20);
  const [breakTimeInSeconds, setBreakTimeInSeconds] = useState(20);
  const [alarmSound, setAlarmSound] = useState(alarmClockSound);
  const [isPlaying, setIsPlaying] = useState(false);

  console.log(runTimeInMinutes, breakTimeInSeconds, alarmSound, isPlaying);

  const onChangeValue = (event) => {
    setAlarmSound(event.target.value);
    console.log(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    setIsPlaying(true);

    console.log(CountdownCircleTimer.props);
  };

  const handleReset = (event) => {
    event.preventDefault();
    setRunTimeInMinutes(20);
    setBreakTimeInSeconds(20);
    setKey((prevKey) => prevKey + 1);
    setAlarmSound(alarmClockSound);
    setIsPlaying(false);
  };

  const renderTime = ({ remainingTime }) => {
    console.log(remainingTime);
    function playSoundEffect() {
      console.log(alarmSound);
      new Audio(alarmSound).play();
    }

    if (remainingTime === 0 && isPlaying) {
      playSoundEffect();
      return (
        <div className="timer">
          <img src={eyeAnimationLookingUp} alt="pair of eyes looking up"></img>
          <p>Look at something far away!</p>
        </div>
      );
    }

    return (
      <div className="timer">
        <img
          src={eyeAnimationSideToSide}
          alt="pair of eyes looking side to side"
        ></img>
      </div>
    );
  };

  return (
    <div className="timer">
      <div className={"timer-instructions"}>
        <h4>Set screen break reminders</h4>
        <p>
          Take frequent breaks from your screen. Your eyes can become tired when
          you use them for long periods. Taking frequent breaks from the screen
          can help to avoid tired eyes and headaches.
          <p>
            The 20, 20, 20 rule suggests taking a break of at least 20 seconds,
            every 20 minutes and to look at least 20 feet away.
          </p>
        </p>
      </div>

      {/* <p>insert YouTube video link here</p> */}
      <div className="timer-wrapper">
        <CountdownCircleTimer
          isPlaying={isPlaying}
          duration={runTimeInMinutes * 60}
          key={key}
          size={350}
          colors={["#8a5a7e", "#F7B801", "#A30000", "#A30000"]}
          //   colors={["#004777", "#F7B801", "#A30000", "#A30000"]}
          colorsTime={[10, 6, 3, 0]}
          onComplete={() => ({ shouldRepeat: true, delay: breakTimeInSeconds })}
        >
          {renderTime}
        </CountdownCircleTimer>
      </div>
      <div className={"timer-setting"}>
        <form>
          <p>
            <b>Set timer here:</b>
          </p>
          <label htmlFor="run_time">Run time (minutes)</label>
          <input
            value={runTimeInMinutes}
            onChange={(event) => setRunTimeInMinutes(event.target.value)}
            name="run_time"
            type="number"
            required
          ></input>
          <label htmlFor="break_time">Break time (seconds)</label>
          <input
            value={breakTimeInSeconds}
            onChange={(event) => setBreakTimeInSeconds(event.target.value)}
            name="break_time"
            type="number"
            required
          ></input>
          <label>
            Choose your alarm sound:
            <select value={alarmSound} onChange={onChangeValue}>
              <option value={alarmClockSound}>Alarm clock</option>
              <option value={birdSound}>Bird song</option>
              <option value={cheeringCrowdSound}>Cheering crowd</option>
              <option value={motivationalMusicSound}>Motivational music</option>
              <option value={pianoSound}>Piano playing</option>
              <option value={soothingMusicSound}>Soothing music</option>
            </select>
          </label>
          <button type="submit" onClick={handleSubmit}>
            Start timer
          </button>
          <button onClick={handleReset}>Reset</button>
        </form>
      </div>
    </div>
  );
}

export default EyeTimer;
