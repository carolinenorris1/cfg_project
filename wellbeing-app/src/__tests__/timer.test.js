import React from "react";
import renderer from "react-test-renderer";

import Timer from "../carousel/Timer/Timer";

it("renders correctly when user views the eye timer on the carousel", () => {
  const tree = renderer.create(<Timer />).toJSON();
  expect(tree).toMatchSnapshot();
});
