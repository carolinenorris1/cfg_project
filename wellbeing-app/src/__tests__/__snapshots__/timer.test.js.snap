// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when user views the eye timer on the carousel 1`] = `
<div
  className="timer"
>
  <div
    className="timer-instructions"
  >
    <h4>
      Set screen break reminders
    </h4>
    <p>
      Take frequent breaks from your screen. Your eyes can become tired when you use them for long periods. Taking frequent breaks from the screen can help to avoid tired eyes and headaches.
      <p>
        The 20, 20, 20 rule suggests taking a break of at least 20 seconds, every 20 minutes and to look at least 20 feet away.
      </p>
    </p>
  </div>
  <div
    className="timer-wrapper"
  >
    <div
      style={
        Object {
          "height": 350,
          "position": "relative",
          "width": 350,
        }
      }
    >
      <svg
        height={350}
        viewBox="0 0 350 350"
        width={350}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m 175,6 a 169,169 0 1,0 0,338 a 169,169 0 1,0 0,-338"
          fill="none"
          stroke="#d9d9d9"
          strokeWidth={12}
        />
        <path
          d="m 175,6 a 169,169 0 1,0 0,338 a 169,169 0 1,0 0,-338"
          fill="none"
          stroke="#8a5a7e"
          strokeDasharray={1061.85831691335}
          strokeDashoffset={0}
          strokeLinecap="round"
          strokeWidth={12}
        />
      </svg>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <div
          className="timer"
        >
          <img
            alt="pair of eyes looking side to side"
            src="eye-animation-side-to-side.gif"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="timer-setting"
  >
    <form>
      <p>
        <b>
          Set timer here:
        </b>
      </p>
      <label
        htmlFor="run_time"
      >
        Run time (minutes)
      </label>
      <input
        name="run_time"
        onChange={[Function]}
        required={true}
        type="number"
        value={20}
      />
      <label
        htmlFor="break_time"
      >
        Break time (seconds)
      </label>
      <input
        name="break_time"
        onChange={[Function]}
        required={true}
        type="number"
        value={20}
      />
      <label>
        Choose your alarm sound:
        <select
          onChange={[Function]}
          value="mechanical-alarm-clock-trimmed.mp3"
        >
          <option
            value="mechanical-alarm-clock-trimmed.mp3"
          >
            Alarm clock
          </option>
          <option
            value="birds-singing-trimmed.mp3"
          >
            Bird song
          </option>
          <option
            value="crowd-cheer-trimmed.mp3"
          >
            Cheering crowd
          </option>
          <option
            value="motivational-trimmed.mp3"
          >
            Motivational music
          </option>
          <option
            value="the-last-piano-trimmed.mp3"
          >
            Piano playing
          </option>
          <option
            value="cottagecore-trimmed.mp3"
          >
            Soothing music
          </option>
        </select>
      </label>
      <button
        onClick={[Function]}
        type="submit"
      >
        Start timer
      </button>
      <button
        onClick={[Function]}
      >
        Reset
      </button>
    </form>
  </div>
</div>
`;
